src/main/java/com/insightfullogic/honest_profiler/core/sources/VirtualMachine.java

```diff
@@ -63,6 +66,19 @@ public boolean isAgentLoaded()
         return agentLoaded;
     }
 
+    public LogSource getLogSourceFromVmArgs()
+    {
+        int agentPathStart = vmArgs.indexOf("-agentpath") + "-agentpath".length();
+        int agentPathEnd = vmArgs.indexOf(' ', agentPathStart);
+        String agentPath = vmArgs.substring(agentPathStart, agentPathEnd);
+        int logPathStart = agentPath.indexOf("logPath=") + "logPath=".length();
+        int commaPos = agentPath.indexOf(",", logPathStart);
+        int spacePos = agentPath.indexOf(' ', logPathStart);
+        int logPathEnd = commaPos > 0 ? commaPos : (spacePos > 0 ? spacePos : agentPath.length());
+
+        return new FileLogSource(new File(agentPath.substring(logPathStart, logPathEnd)));
+    }
```

Comment:
Hm, my bad the. I'll check and fix. The intention was to revert to the original logic if no logpath parameter was found, so as to only extend the number of cases in which the mechanism works correctly.